### PR TITLE
Remove duplicated ILLink PackageReference and update target version of the SDK to 6.0

### LINKED ISSUE #44578: Revert hardcoded DotNetCliVersion in src/tests/Common/helixpublishwitharcade.proj
Revert changes in #44554 once after we update our SDK to use 5.0.0 in our global.json.

### LINKED ISSUE #46960: Start dogfooding .NET 6 SDK/runtime in CI?
I was looking at global.json to see whether we were using a version of SDK which included a specific Apple Silicon fix.  I noticed we are still building with the .NET 5 SDK.  

We should move to .NET 6 SDK before we release .NET 6 preview1 IMHO.

@dotnet/runtime-infrastructure

### LINKED ISSUE #48409: ILLink SDK is defined in multiple places which causes Arcade validation builds to fail
With https://github.com/dotnet/arcade/commit/023bcee5e5f291c5a0fbcbb735a0b3845cc72316, the ILLinker is now referenced globally, similar to the compiler package. As the SDK doesn't allow to have multiple PackageReferences of the same Identitiy when `IsImplicitlyDefined` is set to true, the PackageReference in illink.targets should be removed.

This will unblock the Arcade validation builds: https://dev.azure.com/dnceng/internal/_build/results?buildId=998169&view=logs&j=316d5c15-0c50-544e-8051-e6b14a1ab674&t=68de0bee-b0ee-5e79-8d1a-c77a89225a17

```
C:\hostedtoolcache\windows\dotnet\sdk\5.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(257,5): error NETSDK1023: A PackageReference for 'Microsoft.NET.ILLink.Tasks' was included in your project. This package is implicitly referenced by the .NET SDK and you do not typically need to reference it from your project. For more information, see https://aka.ms/sdkimplicitrefs [D:\a\1\s\Build.proj]
##[error]C:\hostedtoolcache\windows\dotnet\sdk\5.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(257,5): error NETSDK1023: (NETCORE_ENGINEERING_TELEMETRY=Restore) A PackageReference for 'Microsoft.NET.ILLink.Tasks' was included in your project. This package is implicitly referenced by the .NET SDK and you do not typically need to reference it from your project. For more information, see https://aka.ms/sdkimplicitrefs
C:\hostedtoolcache\windows\dotnet\sdk\5.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(257,5): error NETSDK1023: A PackageReference for 'Microsoft.NET.ILLink.Tasks' was included in your project. This package is implicitly referenced by the .NET SDK and you do not typically need to reference it from your project. For more information, see https://aka.ms/sdkimplicitrefs [D:\a\1\s\Build.proj]
##[error]C:\hostedtoolcache\windows\dotnet\sdk\5.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(257,5): error NETSDK1023: (NETCORE_ENGINEERING_TELEMETRY=Restore) A PackageReference for 'Microsoft.NET.ILLink.Tasks' was included in your project. This package is implicitly referenced by the .NET SDK and you do not typically need to reference it from your project. For more information, see https://aka.ms/sdkimplicitrefs
C:\hostedtoolcache\windows\dotnet\sdk\5.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(257,5): error NETSDK1023: A PackageReference for 'Microsoft.NET.ILLink.Tasks' was included in your project. This package is implicitly referenced by the .NET SDK and you do not typically need to reference it from your project. For more information, see https://aka.ms/sdkimplicitrefs [D:\a\1\s\Build.proj]
```

### PR DESCRIPTION
**Pull request by ViktorHofer**
The Arcade.SDK already package refrences the ILLink package. The
duplicate reference in illink.targets caused SDK errors as the Arcade
reference has IsImplicitlyDefined set which doesn't allow an additional
reference with the same identity.

Also, as the ILLink package already exposes the path to the assembly via
its props file, using that instead of manually constructing the path to
the assembly.

The SDK target version update is required as the sequencing of the
ILLink.props file was wrong and is required for this change. This isn't
considered a breaking change, as the SDK's minimum required version
isn't changed.

Fixes https://github.com/dotnet/runtime/issues/48409
Fixes https://github.com/dotnet/runtime/issues/44578
Fixes https://github.com/dotnet/runtime/issues/46960

cc @missymessa

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "rollForward": "major"
   },
   "tools": {
-    "dotnet": "5.0.100"
+    "dotnet": "6.0.100-preview.2.21118.7"
```
Comment (eerhardt) [inline global.json:8]: ?? Is this intended?

Comment (ViktorHofer) [reply global.json:8]: Yes, see the commit description :) It is necessary as the IL.SDK import sequence was wrong and causes problems.

Comment (ViktorHofer) [reply global.json:8]: More precisely, we need this fix: https://github.com/dotnet/sdk/commit/309c55fbf841285acb5772090f65de3916dee363

File: eng/testing/linker/project.csproj.template [hunk 2]
```diff
@@ -28,21 +28,11 @@
 
   <!-- Logic to override the default IlLink tasks that come from the SDK and use the one
   we use in dotnet/runtime repo -->
-
-  <PropertyGroup>
-    <_ILLinkTasksDirectoryRoot>$(PkgMicrosoft_NET_ILLink_Tasks)\tools\</_ILLinkTasksDirectoryRoot>
```
Comment (joperezr) [inline eng/testing/linker/project.csproj.template:33]: The whole point of having this was in order to be able to override the tasks assembly to be able to pick a newer version instead, but I don't think that will be required any longer, so theoretically we could just remove the PackageReference from this project and let the SDK reference the version specified there right? @eerhardt is there still any value of referencing a newer version of the ILLink tasks than the one with the SDK?

File: eng/illink.targets [hunk 3]
```diff
@@ -26,9 +26,6 @@
   <!-- Inputs and outputs of ILLinkTrimAssembly -->
   <PropertyGroup>
     <ILLinkDirectory Condition="'$(ILLinkDirectory)' == ''">$(MSBuildProjectDirectory)\ILLink\</ILLinkDirectory>
-    <ILLinkTasksDir>$([MSBuild]::NormalizeDirectory('$(PkgMicrosoft_NET_ILLink_Tasks)', 'tools'))</ILLinkTasksDir>
```
Comment (joperezr) [inline eng/illink.targets:29]: Should you also remove the GeneratePathProperty from the above packageReference?

File: eng/testing/linker/project.csproj.template (hunk 2 above)
Comment (joperezr) [reply eng/testing/linker/project.csproj.template:33]: mm nvm looks like this package is also getting us the new linker so we might still want to keep the package reference.

Comment (ViktorHofer) [reply eng/testing/linker/project.csproj.template:33]: > The whole point of having this was in order to be able to override the tasks assembly to be able to pick a newer version instead

This is still happening. The IL.SDK's props file now overrides the property that points to the task assembly: https://github.com/mono/linker/blob/422f2f710707c1cf95dd513022c8fbef19f177dc/src/ILLink.Tasks/Sdk/Sdk.props#L15-L16

File: eng/illink.targets (hunk 3 above)
Comment (ViktorHofer) [reply eng/illink.targets:29]: Oops, my mistake. The above PackageReference needs to be removed. Just fixed that.

**Comment by ViktorHofer** [comment]
Waiting for https://github.com/mono/linker/pull/1842 to be merged so that I can consume the newer package.

**Comment by ViktorHofer** [comment]
@jkoritzinsky @hoyosjs can you please take a look at the coreclr error? Seems like the runtime hardcodes the 5.0.0 shared framework somewhere?

**Comment by hoyosjs** [comment]
@ViktorHofer Yes, there are projects that run on 5.0. Namely:

<details>
<summary>References to `NetCoreAppToolCurrent`</summary>

```xml
Directory.Build.props
42:    <NetCoreAppToolCurrentVersion>5.0</NetCoreAppToolCurrentVersion>
47:    <NetCoreAppToolCurrent>net$(NetCoreAppToolCurrentVersion)</NetCoreAppToolCurrent>
48:    <NetCoreAppCurrentToolTargetFrameworkMoniker>$(NetCoreAppCurrentIdentifier),Version=v$(NetCoreAppToolCurrentVersion)</NetCoreAppCurrentToolTargetFrameworkMoniker>
67:    <AppleAppBuilderDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'AppleAppBuilder', 'Debug', '$(NetCoreAppToolCurrent)'))</AppleAppBuilderDir>
68:    <AndroidAppBuilderDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'AndroidAppBuilder', 'Debug', '$(NetCoreAppToolCurrent)', 'publish'))</AndroidAppBuilderDir>
69:    <WasmAppBuilderDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'WasmAppBuilder', 'Debug', '$(NetCoreAppToolCurrent)', 'publish'))</WasmAppBuilderDir>
70:    <WasmBuildTasksDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'WasmBuildTasks', 'Debug', '$(NetCoreAppToolCurrent)', 'publish'))</WasmBuildTasksDir>
71:    <MonoAOTCompilerDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'MonoAOTCompiler', 'Debug', '$(NetCoreAppToolCurrent)'))</MonoAOTCompilerDir>

src/coreclr/crossgen-corelib.proj
4:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/runtime.proj
4:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/ToolBox/SOS/DacTableGen/DacTableGen.csproj
4:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/ToolBox/SOS/DIALib/DIALib.ilproj
3:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/tools/aot/crossgen2/crossgen2.csproj
6:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/tools/aot/ILCompiler.Diagnostics/ILCompiler.Diagnostics.csproj
7:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/tools/aot/ILCompiler.ReadyToRun/ILCompiler.ReadyToRun.csproj
5:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/tools/aot/ILCompiler.TypeSystem.ReadyToRun.Tests/ILCompiler.TypeSystem.ReadyToRun.Tests.csproj
4:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/tools/Common/JitInterface/ThunkGenerator/ThunkGenerator.csproj
5:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/tools/dotnet-pgo/dotnet-pgo.csproj
6:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/tools/r2rdump/R2RDump.csproj
10:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/tools/r2rtest/R2RTest.csproj
6:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/coreclr/tools/runincontext/runincontext.csproj
4:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/installer/tests/Directory.Build.props
12:    <TestInfraTargetFramework>$(NetCoreAppToolCurrent)</TestInfraTargetFramework>

src/mono/llvm/llvm-init.proj
3:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/mono/monoaotcross.proj
3:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/mono/nuget/Microsoft.NET.Runtime.Android.Sample.Mono/Microsoft.NET.Runtime.Android.Sample.Mono.pkgproj
14:    <PackageFile Include="@(_AndroidSampleFiles)" TargetPath="tools\$(NetCoreAppToolCurrent)\" />

src/mono/nuget/Microsoft.NET.Runtime.iOS.Sample.Mono/Microsoft.NET.Runtime.iOS.Sample.Mono.pkgproj
17:    <PackageFile Include="@(_iOSSampleFiles)" TargetPath="tools\$(NetCoreAppToolCurrent)\" />

src/mono/nuget/Microsoft.NET.Runtime.wasm.Sample.Mono/Microsoft.NET.Runtime.wasm.Sample.Mono.pkgproj
14:    <PackageFile Include="@(_wasmSampleFiles)" TargetPath="tools\$(NetCoreAppToolCurrent)\" />

src/mono/sample/Android/AndroidSampleApp.csproj
5:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/mono/sample/HelloWorld/HelloWorld.csproj
4:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/mono/sample/iOS/Program.csproj
5:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/mono/sample/mbr/console/ConsoleDelta.csproj
4:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/mono/sample/mbr/DeltaHelper/DeltaHelper.csproj
3:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/mono/wasm/build/WasmApp.InTree.props
8:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/mono/wasm/build/WasmApp.LocalBuild.props
23:    <_NetCoreAppToolCurrent>net5.0</_NetCoreAppToolCurrent>
32:    <WasmAppBuilderDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'WasmAppBuilder', 'Debug', '$(_NetCoreAppToolCurrent)', 'publish'))</WasmAppBuilderDir>
33:    <WasmBuildTasksDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'WasmBuildTasks', 'Debug', '$(_NetCoreAppToolCurrent)', 'publish'))</WasmBuildTasksDir>
34:    <MonoAOTCompilerDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'MonoAOTCompiler', 'Debug', '$(_NetCoreAppToolCurrent)'))</MonoAOTCompilerDir>

src/mono/wasm/wasm.proj
4:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/tasks/AndroidAppBuilder/AndroidAppBuilder.csproj
3:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/tasks/AotCompilerTask/MonoAOTCompiler.csproj
3:    <TargetFrameworks>$(NetCoreAppToolCurrent)</TargetFrameworks>
29:      <FilesToPackage Include="$(OutputPath)$(NetCoreAppToolCurrent)\$(AssemblyName).dll" TargetPath="tasks" />

src/tasks/AppleAppBuilder/AppleAppBuilder.csproj
3:    <TargetFrameworks>$(NetCoreAppToolCurrent)</TargetFrameworks>

src/tasks/WasmAppBuilder/WasmAppBuilder.csproj
3:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/tasks/WasmBuildTasks/WasmBuildTasks.csproj
3:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/tests/Common/Coreclr.TestWrapper/Coreclr.TestWrapper.csproj
6:    <NuGetTargetMonikerShort>$(NetCoreAppToolCurrent)</NuGetTargetMonikerShort>

src/tests/Common/CoreCLRTestLibrary/CoreCLRTestLibrary.csproj
8:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/tests/Common/dir.sdkbuild.props
9:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/tests/Common/Directory.Build.targets
137:          Include="$(ArtifactsBinDir)\WasmAppBuilder\Debug\$(NetCoreAppToolCurrent)\publish\**"

src/tests/Common/external/external.csproj
15:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/tests/Common/test_dependencies/test_dependencies.csproj
5:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/tests/Common/test_dependencies_fs/test_dependencies.fsproj
5:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>

src/tests/Directory.Build.props
133:    <NuGetTargetMonikerShort>$(NetCoreAppToolCurrent)</NuGetTargetMonikerShort>

src/tests/Directory.Build.targets
359:          Include="$(ArtifactsBinDir)\WasmAppBuilder\Debug\$(NetCoreAppToolCurrent)\publish\**"
364:          Include="$(ArtifactsBinDir)\MonoAOTCompiler\Debug\$(NetCoreAppToolCurrent)\**"

src/tests/JIT/Directed/tailcall/mutual_recursion.fsproj
10:    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>
```

Then just a few questions, as I don't know if it's safe to just remap `NetCoreAppToolCurrent` to `net6.0`:
1. Do we want to build and run tools we use for the product against a 6.0 version?
2. I don't know about a lot of the tools that reference the net5.0 TFM. A bunch of them are mono tasks (`AppleAppBuilder`, `WasmAppBuilder`) and samles (like `AndroidSampleApp`). @directhex are those safe to move to net6.0?
3. @trylek @davidwrighton @AntonLapounov is it safe to move the crossgen2 and dotnet-pgo projects to net6.0?

**Comment by ViktorHofer** [comment]
Yes, I realized that after posting the comment... The tasks are definitely fine to be upgraded. I think crossgen2 should be fine as well.

**Comment by ViktorHofer** [comment]
Apparently, they way that coreclr uses the pkgproj infrastructure causes these failures. Likely because the props and targets files are imported directly there:

```
.packages/microsoft.dotnet.build.tasks.packaging/6.0.0-beta.21120.1/build/Packaging.targets(747,5): error MSB4018: (NETCORE_ENGINEERING_TELEMETRY=Build) The "GenerateRuntimeDependencies" task failed unexpectedly.
System.MissingMethodException: Method not found: 'Void NuGet.RuntimeModel.JsonObjectWriter..ctor(Newtonsoft.Json.JsonWriter)'.
   at Microsoft.DotNet.Build.Tasks.Packaging.NuGetUtility.WriteRuntimeGraph(String filePath, RuntimeGraph runtimeGraph)
   at Microsoft.DotNet.Build.Tasks.Packaging.GenerateRuntimeDependencies.Execute() in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/GenerateRuntimeDependencies.cs:line 144
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)
```

@ericstj can you please help out here? I also updated the NuGet dependencies in Arcade and they already flowed into this PR. The binlog shows that the right assembly with the NuGet breaking change is used. It's unclear to me why there's a MissingMethodException.

**Comment by ViktorHofer** [comment]
OK I got the fix ready in Arcade, and will update this PR tomorrow.

File: eng/Tools.props [hunk 4]
```diff
@@ -18,4 +18,14 @@
...
   </ItemGroup>
 
+  <!-- Mega hack to remove Newtonsoft.Json dependency which doesn't unify during runtime with the dependency in the SDK and causes MissingMethodExceptions. -->
+  <Target Name="RemovePackagingContentForMSBuildALC"
+          AfterTargets="Restore">
+    <PropertyGroup>
+      <_BuildTasksPackagingNewtonsoftDependency>$(NuGetPackageRoot)microsoft.dotnet.build.tasks.packaging\$(MicrosoftDotNetBuildTasksPackagingVersion)\tools\netcoreapp3.1\Newtonsoft.Json.dll</_BuildTasksPackagingNewtonsoftDependency>
+    </PropertyGroup>
+
+    <Delete Files="$(_BuildTasksPackagingNewtonsoftDependency)" Condition="Exists('$(_BuildTasksPackagingNewtonsoftDependency)')" />
```
Comment (eerhardt) [inline eng/Tools.props:28]: ick. Do we have an issue logged to track fixing this for real? This seems broken.

File: src/coreclr/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets [hunk 5]
```diff
@@ -30,7 +26,7 @@
         of child processes, and attach a debugger to msbuild. After that you can set a bp on the exception
         you are seeing.
   -->
-  <UsingTask TaskName="CreateRuntimeRootILLinkDescriptorFile" AssemblyFile="$(_ILLinkTasksPath)" />
+  <UsingTask TaskName="CreateRuntimeRootILLinkDescriptorFile" AssemblyFile="$(ILLinkTasksAssembly)" Condition="'$(ILLinkTasksAssembly)' != ''" />
```
Comment (eerhardt) [inline src/coreclr/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets:29]: For corelib, is there ever a case where `'$(ILLinkTasksAssembly)' == ''`?

File: src/installer/tests/Assets/TestUtils/TestProjects.targets [hunk 6]
```diff
@@ -3,29 +3,9 @@
     This file is imported by the test projects from the artifacts dir or the src/tests dir. It
     provides basic info needed for restore and build with the vanilla SDK.
   -->
-  <PropertyGroup>
-    <NETCoreAppMaximumVersion>6.0</NETCoreAppMaximumVersion>
-  </PropertyGroup>
+
   <!--
     We are using to best emulate the live version. This should be removed once we start using the live bits.
```
Comment (eerhardt) [inline src/installer/tests/Assets/TestUtils/TestProjects.targets:8]: Can we remove this comment now?

Can the whole file be removed?

File: src/coreclr/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets (hunk 5 above)
Comment (ViktorHofer) [reply src/coreclr/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets:29]: For any project which hasn't restored yet and included the nuget generated properties and targets files. `ILLinkTasksAssembly` isn't available before that nuget.g.props file is imported. As MSBuild already evaluates UsingTasks before they are used, the `AssemblyFile` parameter must not be empty, hence the condition.

File: src/installer/tests/Assets/TestUtils/TestProjects.targets (hunk 6 above)
Comment (ViktorHofer) [reply src/installer/tests/Assets/TestUtils/TestProjects.targets:8]: Nope, that issue is still accurate, we need to use the live built targeting pack to build the installer tests. Using a prebuilt here seems unnecessary and just adds a prebuilt dependency to our build.

Comment (eerhardt) [reply src/installer/tests/Assets/TestUtils/TestProjects.targets:8]: I don't understand. This file only contains comments. What value is it serving?

Comment (ViktorHofer) [reply src/installer/tests/Assets/TestUtils/TestProjects.targets:8]: installer tests don't import the rest of the repo's infrastructure. That file is the point where properties or targets should live which are necessary for the installer tests. That file was already exactly in the current state (empty) in 5.0 before the KnownFrameworkReference was added, just without the comment.

File: eng/Tools.props (hunk 4 above)
Comment (ViktorHofer) [reply eng/Tools.props:28]: Not yet but hopefully later today. Will need some time to write that issue down in dotnet/msbuild.

**Comment by ViktorHofer** [comment]
@trylek @jkoritzinsky @hoyosjs can someone pls help me figure out why all runtime tests blow up with a 6.0 SDK? I don't even know how to repro this locally. Currently looking into the correlation payloads...

**Comment by ViktorHofer** [comment]
Oh wow, I just noticed that the SDK value is hardcoded for the runtime tests to 5.0: https://github.com/dotnet/runtime/blob/2efa7757ee83a5cb5444a7e44a9687a3c5e2ea3a/src/tests/Common/helixpublishwitharcade.proj#L97.

Questions that I have:
- Why do we require the runtime to be downloaded for the runtime tests? Aren't the runtime tests using the live built shared framework?
- Why is that value hardcoded and doesn't use the value from the repo's global.json as done below for mobile tests?

**Comment by jkoritzinsky** [comment]
The runtime tests themselves run on the built shared framework, but the xunit wrappers around the tests run on the CLI we pull down. Otherwise when something breaks that xunit uses, we lose all test coverage, which makes the problem harder to track down.

**Comment by jkoritzinsky** [comment]
The xunit wrappers execute the test and assert the return code matches the expected value. Without them, we wouldn’t have a good way to execute our tests and get an xunit test results file to upload to AzDO. We could use a different runner or create a new one (which we do for onboarding new targets), but the xunit system works well for now.

File: eng/Tools.props (hunk 4 above)
Comment (ViktorHofer) [reply eng/Tools.props:28]: Eric filed one: https://github.com/dotnet/msbuild/issues/6186

**Comment by ViktorHofer** [comment]
@safern @akoeplinger I think you guys touched the code that handles the xunit asset publishing with xharness. Can you please take a look at the error?

```
/__w/1/s/.dotnet/sdk/6.0.100-preview.2.21118.7/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.ConflictResolution.targets(108,5): error NETSDK1148: Found multiple publish output files with the same relative path: /__w/1/s/artifacts/bin/Microsoft.XmlSerializer.Generator.Tests/net6.0-Release/browser-wasm/Microsoft.XmlSerializer.Generator.Tests.deps.json, /__w/1/s/artifacts/obj/Microsoft.XmlSerializer.Generator.Tests/net6.0-Release/browser-wasm/Microsoft.XmlSerializer.Generator.Tests.deps.json, /__w/1/s/.packages/xunit.abstractions/2.0.3/lib/netstandard2.0/xunit.abstractions.dll, /__w/1/s/artifacts/bin/WasmTestRunner/net6.0-Release/xunit.abstractions.dll, /__w/1/s/.packages/xunit.extensibility.core/2.4.1/lib/netstandard1.1/xunit.core.dll, /__w/1/s/artifacts/bin/WasmTestRunner/net6.0-Release/xunit.core.dll, /__w/1/s/.packages/xunit.extensibility.execution/2.4.1/lib/netstandard1.1/xunit.execution.dotnet.dll, /__w/1/s/artifacts/bin/WasmTestRunner/net6.0-Release/xunit.execution.dotnet.dll. [/__w/1/s/src/libraries/Microsoft.XmlSerializer.Generator/tests/Microsoft.XmlSerializer.Generator.Tests.csproj]
```

**Comment by ViktorHofer** [comment]
The remaining issue are duplicated xunit artifacts in mobile builds that @safern is currently looking at. I expect (or better to say hope) that we get this in by EOD.

**Review by safern — APPROVED** [review]
Seems reasonable to me.

**Review by eerhardt — APPROVED** [review]
Looks good.

I still don't understand why we need the empty `TestProjects.targets` file, but if you think it is valuable, then I'm OK with it.

**Comment by ViktorHofer** [comment]
DNS failures are https://github.com/dotnet/runtime/issues/48751

**Comment by sdmaclea** [comment]
I thought this was supposed to wait until the March infras rollout.  Or am I missing something?

**Comment by ViktorHofer** [comment]
> I thought this was supposed to wait until the March infras rollout. Or am I missing something?

The march rollout updates the SDK's **minimum required version** (the sdk entry in the repo's global.json file). msbuild checks that version when its being invoked (including VS). The target version that was being updated as part of this PR is used for bootstrapping the repo local SDK.

**Pull request merged**